Lib/genericpath.py

```diff
@@ -149,7 +149,7 @@ def _check_arg_types(funcname, *args):
         elif isinstance(s, bytes):
             hasbytes = True
         else:
-            raise TypeError('%s() argument must be str or bytes, not %r' %
-                            (funcname, s.__class__.__name__)) from None
+            raise TypeError(f'{funcname}() argument must be str, bytes, or '
+                            f'os.PathLike object, not {s.__class__.__name__}') from None
```

Comment:
In order to use %r formatting in an f-string, the expression within the braces has to end with an `!r`:
```suggestion
                            f'os.PathLike object, not {s.__class__.__name__!r}') from None
```

Reply:
Good call, thank you!